### ISSUE TITLE
[Bug] MSAL should not clean existing token cache when failed to get token silently for disabled account of one tenant 

### ISSUE THREAD

**Issue by erich-wang** [AzureAD/microsoft-authentication-library-for-dotnet#2535]
**Logs and Network traces**
Without logs or traces, it is unlikely that the team can investigate your issue. Capturing logs and network traces is described at https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/wiki/logging

**Which Version of MSAL are you using ?**

MSAL 4.23

**Platform**

**What authentication flow has the issue?**
* Desktop / Mobile
    * [ ] Interactive
    * [ ] Integrated Windows Auth
    * [ ] Username Password
    * [ ] Device code flow (browserless)
* Web App 
    * [ ] Authorization code
    * [ ] OBO
* Daemon App 
    * [ ] Service to Service calls

Other? - please describe;

**Is this a new or existing app?**

**Repro**

Suppose one user has two tenants, tenant a and tenant b, but the user is disabled in tenant b. What Connect-AzAccount does:
1. Use Interactive to get token for tenant a successfully, token is saved into token cache;
2. Silently to get token for tenant b, fails with 400 status code with detail error below, and existing token for tenant a is deleted.

```
{"error":"invalid_grant","error_description":"AADSTS50057: The user account is disabled.\r\nTrace ID: 669550bf-8a30-4952-a5cf-41b0974fef00\r\nCorrelation ID: cd0d0527-4790-4059-a34d-22f8f2ec9b30\r\nTimestamp: 2021-04-06 01:07:49Z","error_codes":[50057],"timestamp":"2021-04-06 01:07:49Z","trace_id":"669550bf-8a30-4952-a5cf-41b0974fef00","correlation_id":"cd0d0527-4790-4059-a34d-22f8f2ec9b30","error_uri":"https://login.microsoftonline.com/error?code=50057","suberror":"bad_token"}
```
**Expected behavior**
When fails to get token silently for tenant b, it should keep existing token cache.

**Actual behavior**
 When fails to get token silently for tenant b, existing token for tenant a is deleted including refresh token.

**Possible Solution**

**Additional context/ Logs / Screenshots**
Add any other context about the problem here, such as logs and screebshots.

**Comment by bgavrilMS** [comment]
Hi @erich-wang , Visual Studio has also faced this problem and we have fixed the issue in MSAL 4.27. Please use upgrade to latest MSAL and it will work. 

See duplicate issue raised by VS: https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/issues/2294

**Comment by jiasli** [comment]
MSAL.PY faced similar issue that it deletes the refresh token upon receiving `invalid_grant` (https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/315). It is solved by https://github.com/AzureAD/microsoft-authentication-library-for-python/pull/315.
